providers/base/bin/network.py

```diff
@@ -753,24 +767,208 @@ def make_target_list(iface, test_targets, log_warnings):
+        raise SystemExit(
+            "the network speed is incorrect for {} interface".format(
+                target_network
+            )
+        )
+
+    if (
+        target_if_attrs["phys_switch_id"] is not None
+        and target_if_attrs["ifindex"] != target_if_attrs["iflink"]
+    ):
+        # Means it's not a Physical network, but a Linux DSA network
+        # Reference:
+        # https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-class-net
+        conduit_net = [
+            iface
+            for iface, attr in network_info.items()
+            if attr["ifindex"] == target_if_attrs["iflink"]
+        ]
+        if not conduit_net:
+            raise SystemExit("Conduit network interface not found")
+
+        conduit_net = conduit_net[0]
+        conduit_if_attrs = network_info.pop(conduit_net)
+
+        # bring up conduit network interface as well
+        if conduit_if_attrs["status"] == "down" and not turn_up_network(
+            conduit_net, timeout
+        ):
+            raise SystemExit(
+                "Failed to bring up {} conduit interface".format(conduit_net)
+            )
+
+    if toggle_status:
+        # Shutdown other network interfaces
+        for iface, attrs in network_info.items():
+            if attrs["status"] == "up" and not turn_down_network(iface):
+                raise SystemExit(
+                    "Failed to shutdown {} interface".format(iface)
+                )
+
+
+def restore_network_ifaces(cur_network_info, origin_network_info, timeout):
+    status = True
+
+    logging.debug("Restoring interface")
+    for iface, attrs in origin_network_info.items():
+        if attrs["status"] != cur_network_info[iface]["status"]:
+            if attrs["status"] == "up" and not turn_up_network(iface, timeout):
+                status = False
+            elif attrs["status"] == "down" and not turn_down_network(iface):
+                status = False
+
+    return status
+
+
+@contextmanager
+def interface_test_initialize(
+    target_dev, underspeed_ok, dont_toggle_ifaces, recover_timeout
+):
+    tempfile_route = tempfile.TemporaryFile()
+    try:
+        network_info = get_network_ifaces()
+        # Back up routing table, since network down/up process
+        # tends to trash it....
+        logging.debug("Backup routing table")
+        check_call(
+            ["ip", "route", "save", "table", "all"], stdout=tempfile_route
+        )
+        setup_network_ifaces(
+            network_info,
+            target_dev,
+            underspeed_ok,
+            not dont_toggle_ifaces,
+            recover_timeout,
+        )
+
+        yield
+
+    finally:
+        cur_network_info = get_network_ifaces()
+        recover_success = restore_network_ifaces(
+            cur_network_info, network_info, recover_timeout
+        )
+
+        # Restore routing table to original state
+        logging.debug("Restore routing table")
+        try:
```

Comment:
Like the comment! ty, I would use `with suppress(CalledProcessError)` or alternatively... `run` istead of `check_call` there is no point to using `check_call` if you ignore the check!

Reply:
thanks for the suggestion, I have applied it into the latest commit